nncf/torch/tensor_statistics/statistics.py

```diff
@@ -18,25 +20,63 @@
 from nncf.common.tensor_statistics.statistics import TensorStatistic
 
 
+def _reshape_all(targets: Tuple[torch.Tensor, ...], target_shape: Tuple[int, ...]):
+    return map(lambda stat: torch.reshape(stat, target_shape), targets)
+
+
 class PTMinMaxTensorStatistic(MinMaxTensorStatistic):
```

Comment:
Let's discuss it offline. I don't think we need to make aggregators output strict, I think it's good idea to allow returning anything and then interpretate it by TensorStatistic.

Reply:
Could you please give me an example on how should I refactor aggregators output? For now aggregators return `Dict[str, Tensor]`, `Tensor`, `Tuple[int, ...]` and so on, in what place should I unify them?

Other consideration is that for some Statistics we need several aggregators (like MinMax (Min + Max), Mean(Mean, Shape)) and for some Statistics we need only one aggregator (Percentile for example)

Reply:
If this refactoring requires a lot of changes (>100 lines of code) in the code that was introduced before this PR (Aggregators, for example) I will be happy to refactor the code to make it clearer, but in a separate PR

Reply:
This is a serious architectural task that, in fact, should have been considered ahead of introducing the reducer-aggregator solution, and it relates to the issues I brought up before, for example, the inconsistency of `NoopAggregator`'s return type. 

Your solution makes the derivation of a statistic from a collector something that is determined in runtime, not statically by code. There can now be a situation when a given collector has valid reducers and aggregators, but the collection results cannot be turned into a valid statistic. This is a feature of the solution, but this means that there must be additional, runtime logic that can handle a request to the collector in order to get a statistic of a specific kind, and maybe fail. I presume that somewhere in the implementation a [visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern) must be used, although in Python it can probably, in all honesty, replaced by a chain of `if isinstance(...)` calls without significant difference in perf. Implementing a visitor pattern explicitly, however, will allow you to do proper type hinting and make the solution type-correct.

Reply:
Visitors is a nice idea! Do you insist we should do this in this PR or can we move it to a refactoring?

Other idea I was discussing with @alexsu52 is that PTQ interface does not need any post processing, only backend that requires this is original Torch backend. As original torch quantization algorithm is planned to be refactored to be PTQ compatible, this reshape should eventual gone. The point is that all code that will be reshaping tensors by visitors should be abandoned at some point, thus, maybe visitors interface is a potentially an overhead for this task

Reply:
All I'm asking you to do right now is to do the tensor collector output unpacking at the point where you instantiate these classes, not inside them.

Reply:
Done, please check